pkg/deploy/reaper/reaper.go

```diff
@@ -35,9 +36,21 @@ type DeploymentConfigReaper struct {
 // zero replicas, waits for all of them to get deleted and then deletes both the
 // replication controller and its deployment configuration.
 func (reaper *DeploymentConfigReaper) Stop(namespace, name string, gracePeriod *kapi.DeleteOptions) (string, error) {
-	if err := reaper.osc.DeploymentConfigs(namespace).Delete(name); err != nil {
+	// If the config is already deleted, it may still have associated
+	// deployments which didn't get cleaned up during prior calls to Stop. If
+	// the config can't be found, still make an attempt to clean up the
+	// deployments.
+	//
+	// It's important to delete the config first to avoid an undesirable side
+	// effect which can cause the deployment to be re-triggered upon the
+	// config's deletion. See https://github.com/openshift/origin/issues/2721
+	// for more details.
+	err := reaper.osc.DeploymentConfigs(namespace).Delete(name)
+	if err != nil && !kerrors.IsNotFound(err) {
```

Comment:
if the deployment config is not found and there are no deployments, the NotFound error should be returned